src/test/java/com/adyen/PaymentTest.java

```diff
@@ -477,16 +459,80 @@ protected void assertRefused(PaymentResult paymentResult) {
     }
 
     @Test
-    @Ignore
+    public void TestByteArrayDeserialization() throws Exception {
+
+        Client client = createMockClientFromFile("mocks/authorise-success.json");
+        Payment payment = new Payment(client);
+
+        final String expectedBytesAsString = "Let's pretend this a jpg or something";
+        final byte[] expectedBytes = expectedBytesAsString.getBytes(StandardCharsets.UTF_8);
+        final ByteString expectedByteString = ByteString.of(expectedBytes);
+        final String serializedBytes = expectedByteString.base64();
+        final String serializedBytesWithQuotes = "\"" + serializedBytes + "\"";
+        Type type = new TypeToken<byte[]>() { }.getType();
+
+        // Act
+        byte[] actualDeserializedBytes = JSON.deserialize(serializedBytesWithQuotes, type);
+
+        // Assert
+        assertEquals(expectedBytesAsString, new String(actualDeserializedBytes, StandardCharsets.UTF_8));
+    }
+
+    @Test
+    public void TestByteArraySerialization() throws Exception {
+
+        Client client = createMockClientFromFile("mocks/authorise-success.json");
+        Payment payment = new Payment(client);
+
+        final byte[] bytes =  "Let's pretend this a jpg or something".getBytes(StandardCharsets.UTF_8);
+        String serialized = JSON.serialize(bytes);
+        String serialized1 = serialized.substring(1, serialized.length() - 1);
+        if (JSON.getGson().htmlSafe()) {
+            serialized1 = serialized1.replaceAll("\\\\u003d", "=");
+        }
+
+        String expectedBytesAsString = "Let's pretend this a jpg or something";
+
+        ByteString actualAsByteString = ByteString.decodeBase64(serialized1);
+
+        byte[] actualBytes = actualAsByteString.toByteArray();
+
+        assertEquals(expectedBytesAsString, new String(actualBytes, StandardCharsets.UTF_8));
+    }
+    @Test
     public void TestByteArrayToJSONString() throws Exception {
         Client client = createMockClientFromFile("mocks/authorise-success.json");
         Payment payment = new Payment(client);
         PaymentRequest paymentRequest = new PaymentRequest();
-        paymentRequest.mpiData(new ThreeDSecureData().cavv("AQIDBAUGBwgJCgsMDQ4PEBESExQ=".getBytes()));
+        ThreeDSecureData threeDSecureData = new ThreeDSecureData();
+
+        threeDSecureData.setCavv("AQIDBAUGBwgJCgsMDQ4PEBESExQ".getBytes(StandardCharsets.UTF_8));
+        paymentRequest.mpiData(threeDSecureData);
         
         payment.authorise(paymentRequest);
-        
-        String expected = "\"mpiData\":{\"cavv\":\"AQIDBAUGBwgJCgsMDQ4PEBESExQ=\"}";
+
+        // In the json the string is encoded in base64
+        String expected = "\"mpiData\":{\"cavv\":\"QVFJREJBVUdCd2dKQ2dzTURRNFBFQkVTRXhR\"}";
```

Comment:
The previous JSON string expectation is correct: 

```
String expected = "\"mpiData\":{\"cavv\":\"AQIDBAUGBwgJCgsMDQ4PEBESExQ=\"}";
```

The string "AQIDBAUGBwgJCgsMDQ4PEBESExQ=" is already base64 encoded. The string "QVFJREJBVUdCd2dKQ2dzTURRNFBFQkVTRXhR" is another base64 encode (double encoding) of what the user set via:

```
paymentRequest.mpiData(new ThreeDSecureData().cavv("AQIDBAUGBwgJCgsMDQ4PEBESExQ=".getBytes()));
```  

This is happening now because the new [ByteArrayAdapter](https://github.com/Adyen/adyen-java-api-library/blob/1814a41bf7f7509a1a588b3c8d0d921a01b7c32e/src/main/java/com/adyen/model/payments/JSON.java#L222) applies base64 by default assuming the input is not encoded already.  

See also my previous [investigation resources](https://github.com/Adyen/adyen-java-api-library/pull/848#issue-1411156955).